src/emq_web_hook.erl

```diff
@@ -225,15 +225,40 @@ on_message_acked(ClientId, Username, Message = #mqtt_message{topic = Topic}, {Fi
 %% Internal functions
 %%--------------------------------------------------------------------
 
+-spec get_timestamp_ms() -> integer().
+get_timestamp_ms() ->
+    {Mega, Sec, Micro} = os:timestamp(),
+    (Mega*1000000 + Sec)*1000 + round(Micro/1000).
+
 send_http_request(Params) ->
     Params1 = iolist_to_binary(mochijson2:encode(Params)),
     Url = application:get_env(?APP, url, "http://127.0.0.1"),
     ?LOG(debug, "Url:~p, params:~s", [Url, Params1]),
-    case httpc:request(post, {Url, [], "application/json", Params1}, [{timeout, 5000}], []) of
-        {ok, _} -> 
+    case perform_request(post, {Url, [], "application/json", Params1}, [{timeout, 5000}], [], 1, 4) of
+        ok ->
+            ok;
+        {error, Reason} ->
+            ?LOG(error, "HTTP request error: ~p", [Reason]),
+            ok
+    end.
+
+perform_request(Method, Req, HTTPOpts, Opts, Attempt, MaxAttempt) ->
+    StartTime = get_timestamp_ms(),
+    case httpc:request(Method, Req, HTTPOpts, Opts) of
+        {ok, _} ->
+            Tags = #{ pod_name => os:getenv("HOSTNAME"), status => "success", attempt => Attempt },
+            dogstatsd:histogram("mqtt_broker.webhook.latency_ms", get_timestamp_ms() - StartTime, Tags),
             ok;
         {error, Reason} ->
-            ?LOG(error, "HTTP request error: ~p", [Reason]), ok %% TODO: return ok?
+            Tags = #{ pod_name => os:getenv("HOSTNAME"), status => "failure", attempt => Attempt },
```

Comment:
pod_name is not necessary, usually we already have host as a tag. Please remove it. 

Also, we should not force to use this plugin on kubernetes. 

Finally, using as statefulset is ok since the pod names are controlled. But if using in a deployment, the variety of names could generate a problem of too many dimensions of tags on datadog.

Reply:
Marcelo from the SRE team advised me to send the pod_name tag in case of the source discovery feature of our Datadog Agent is disabled, but I agree that relying in the HOSTNAME env var provided by k8s is a bad choice... 

But I don't get the point of the too many dimensions problem. As you pointed, we have a host tag for all metrics... Wouldn't we have this too many dimensions problem already?

Reply:
Done!

Reply:
That's the point, there is a default tag with host, we don't need another one